### ISSUE TITLE
Port PGM

### ISSUE THREAD

**Issue by finger563** [pranav-srinivas-kumar/OpenGL-Engine#1]
Let's get PGM to render the surface of the objects instead of icosahedron subdivision :)

**Comment by pranav-srinivas-kumar** [comment]
Yeah definitely. I'm porting this first to Pure C (for fun of course) :)

**Comment by pranav-srinivas-kumar** [comment]
Yeah definitely. I'm porting this pure C first :)
